api/integration_alert_channels_datadog.go

```diff
@@ -0,0 +1,123 @@
+	DatadogServiceLogsDetails   datadogService = "Logs Detail"
+	DatadogServiceEventsSummary datadogService = "Events Summary"
+	DatadogServiceLogsSummary   datadogService = "Logs Summary"
+)
+
+// NewDatadogAlertChannel returns an instance of DatadogAlertChannel
+// with the provided name and data.
+//
+// Basic usage: Initialize a new DatadogAlertChannel struct, then
+//              use the new instance to do CRUD operations
+//
+//   client, err := api.NewClient("account")
+//   if err != nil {
+//     return err
+//   }
+//
+// datadog := api.NewDatadogAlertChannel("foo",
+//   api.DatadogChannelData{
+// 	  DatadogSite:    api.DatadogSiteEu.String(),
+// 	  DatadogService: api.DatadogServiceEventsSummary.String(),
+// 	  ApiKey:      	"datadog-key",
+//   },
+// )
+//
+//   client.Integrations.CreateDatadogAlertChannel(datadogChannel)
+//
+func NewDatadogAlertChannel(name string, data DatadogChannelData) DatadogAlertChannel {
+	return DatadogAlertChannel{
+		commonIntegrationData: commonIntegrationData{
+			Name:    name,
+			Type:    DatadogChannelIntegration.String(),
+			Enabled: 1,
+		},
+		Data: data,
+	}
+}
+
+// CreateDatadogAlertChannel creates a datadog alert channel integration on the Lacework Server
+func (svc *IntegrationsService) CreateDatadogAlertChannel(integration DatadogAlertChannel) (
+	response DatadogAlertChannelResponse,
+	err error,
+) {
+	err = svc.create(integration, &response)
+	return
+}
+
+// GetDatadogAlertChannel gets a datadog alert channel integration that matches with
+// the provided integration guid on the Lacework Server
+func (svc *IntegrationsService) GetDatadogAlertChannel(guid string) (response DatadogAlertChannelResponse,
+	err error) {
+	err = svc.get(guid, &response)
+	return
+}
+
+// UpdateDatadogAlertChannel updates a single datadog alert channel integration
+func (svc *IntegrationsService) UpdateDatadogAlertChannel(data DatadogAlertChannel) (
+	response DatadogAlertChannelResponse,
+	err error,
+) {
+	err = svc.update(data.IntgGuid, data, &response)
+	return
+}
+
+// ListDatadogAlertChannel lists the datadog alert channel integrations available on the Lacework Server
+func (svc *IntegrationsService) ListDatadogAlertChannel() (response DatadogAlertChannelResponse, err error) {
+	err = svc.listByType(DatadogChannelIntegration, &response)
+	return
+}
+
+func (i datadogSite) String() string {
+	return string(i)
+}
+
+func (i datadogService) String() string {
+	return string(i)
+}
+
+type DatadogAlertChannelResponse struct {
+	Data    []DatadogAlertChannel `json:"data"`
+	Ok      bool                  `json:"ok"`
+	Message string                `json:"message"`
+}
+
+type DatadogAlertChannel struct {
+	commonIntegrationData
+	Data DatadogChannelData `json:"DATA"`
+}
+
+type DatadogChannelData struct {
+	DatadogSite    string `json:"DATADOG_SITE,omitempty" mapstructure:"DATADOG_SITE"`
+	DatadogService string `json:"DATADOG_TYPE,omitempty" mapstructure:"DATADOG_TYPE"`
```

Comment:
I thought we wanted to do this
```suggestion
	DatadogSite    datadogSite    `json:"DATADOG_SITE,omitempty" mapstructure:"DATADOG_SITE"`
	DatadogService datadogService `json:"DATADOG_TYPE,omitempty" mapstructure:"DATADOG_TYPE"`
```